File: Pinball App 2/Pinball App 2/gameroom/GameRoomEnumModels.swift
import Foundation

enum OwnedMachineStatus: String, CaseIterable, Codable, Identifiable {
    case active
    case loaned
    case archived
    case sold
    case traded

    var id: String { rawValue }

    var countsAsActiveInventory: Bool {
        self == .active || self == .loaned
    }
}

enum GameRoomAttentionState: String, CaseIterable, Codable, Identifiable {
    case red
    case yellow
    case green
    case gray

    var id: String { rawValue }
}

enum MachineEventCategory: String, CaseIterable, Codable, Identifiable {
    case service
    case ownership
    case mod
    case media
    case inspection
    case issue
    case custom

    var id: String { rawValue }
}

enum MachineEventType: String, CaseIterable, Codable, Identifiable {
    case glassCleaned
    case playfieldCleaned
    case ballsCleaned
    case ballsReplaced
    case pitchChecked
    case machineLeveled
    case rubbersReplaced
    case flipperServiced
    case generalInspection
    case partReplaced
    case modInstalled
    case modRemoved
    case purchased
    case moved
    case loanedOut
    case returned
    case listedForSale
    case sold
    case traded
    case reacquired
    case issueOpened
    case issueResolved
    case photoAdded
    case videoAdded
    case custom

    var id: String { rawValue }
}

enum MachineIssueStatus: String, CaseIterable, Codable, Identifiable {
    case open
    case monitoring
    case resolved
    case deferred

    var id: String { rawValue }
}

enum MachineIssueSeverity: String, CaseIterable, Codable, Identifiable {
    case low
    case medium
    case high
    case critical

    var id: String { rawValue }
}

enum MachineIssueSubsystem: String, CaseIterable, Codable, Identifiable {
    case flipper
    case slingshot
    case popBumper
    case trough
    case shooterLane
    case switchMatrix
    case opto
    case coil
    case magnet
    case diverter
    case ramp
    case toyMech
    case lighting
    case sound
    case display
    case cabinet
    case software
    case network
    case other

    var id: String { rawValue }
}

enum MachineAttachmentOwnerType: String, CaseIterable, Codable, Identifiable {
    case event
    case issue

    var id: String { rawValue }
}

enum MachineAttachmentKind: String, CaseIterable, Codable, Identifiable {
    case photo
    case video

    var id: String { rawValue }
}

enum MachineReminderTaskType: String, CaseIterable, Codable, Identifiable {
    case glassCleaned
    case playfieldCleaned
    case ballsReplaced
    case pitchChecked
    case machineLeveled
    case rubbersReplaced
    case flipperServiced
    case generalInspection

    var id: String { rawValue }

    var matchingEventTypes: [MachineEventType] {
        switch self {
        case .glassCleaned:
            return [.glassCleaned]
        case .playfieldCleaned:
            return [.playfieldCleaned]
        case .ballsReplaced:
            return [.ballsReplaced]
        case .pitchChecked:
            return [.pitchChecked]
        case .machineLeveled:
            return [.machineLeveled]
        case .rubbersReplaced:
            return [.rubbersReplaced]
        case .flipperServiced:
            return [.flipperServiced]
        case .generalInspection:
            return [.generalInspection]
        }
    }
}

enum MachineReminderMode: String, CaseIterable, Codable, Identifiable {
    case dateBased
    case playBased
    case manualOnly

    var id: String { rawValue }
}

enum MachineImportSource: String, CaseIterable, Codable, Identifiable {
    case pinside

    var id: String { rawValue }
}

enum MachineImportMatchConfidence: String, CaseIterable, Codable, Identifiable {
    case high
    case medium
    case low
    case manual

    var id: String { rawValue }
}

File: Pinball App Android/app/src/main/java/com/pillyliu/pinprofandroid/gameroom/GameRoomEnumModels.kt
package com.pillyliu.pinprofandroid.gameroom

internal enum class OwnedMachineStatus {
    active,
    loaned,
    archived,
    sold,
    traded,
    ;

    val countsAsActiveInventory: Boolean
        get() = this == active || this == loaned
}

internal enum class GameRoomAttentionState {
    red,
    yellow,
    green,
    gray,
}

internal enum class MachineEventCategory {
    service,
    ownership,
    mod,
    media,
    inspection,
    issue,
    custom,
}

internal enum class MachineEventType {
    glassCleaned,
    playfieldCleaned,
    ballsCleaned,
    ballsReplaced,
    pitchChecked,
    machineLeveled,
    rubbersReplaced,
    flipperServiced,
    generalInspection,
    partReplaced,
    modInstalled,
    modRemoved,
    purchased,
    moved,
    loanedOut,
    returned,
    listedForSale,
    sold,
    traded,
    reacquired,
    issueOpened,
    issueResolved,
    photoAdded,
    videoAdded,
    custom,
}

internal enum class MachineIssueStatus {
    open,
    monitoring,
    resolved,
    deferred,
}

internal enum class MachineIssueSeverity {
    low,
    medium,
    high,
    critical,
}

internal enum class MachineIssueSubsystem {
    flipper,
    slingshot,
    popBumper,
    trough,
    shooterLane,
    switchMatrix,
    opto,
    coil,
    magnet,
    diverter,
    ramp,
    toyMech,
    lighting,
    sound,
    display,
    cabinet,
    software,
    network,
    other,
}

internal enum class MachineAttachmentOwnerType {
    event,
    issue,
}

internal enum class MachineAttachmentKind {
    photo,
    video,
}

internal enum class MachineReminderTaskType {
    glassCleaned,
    playfieldCleaned,
    ballsReplaced,
    pitchChecked,
    machineLeveled,
    rubbersReplaced,
    flipperServiced,
    generalInspection,
    ;

    val matchingEventTypes: Set<MachineEventType>
        get() = when (this) {
            glassCleaned -> setOf(MachineEventType.glassCleaned)
            playfieldCleaned -> setOf(MachineEventType.playfieldCleaned)
            ballsReplaced -> setOf(MachineEventType.ballsReplaced)
            pitchChecked -> setOf(MachineEventType.pitchChecked)
            machineLeveled -> setOf(MachineEventType.machineLeveled)
            rubbersReplaced -> setOf(MachineEventType.rubbersReplaced)
            flipperServiced -> setOf(MachineEventType.flipperServiced)
            generalInspection -> setOf(MachineEventType.generalInspection)
        }
}

internal enum class MachineReminderMode {
    dateBased,
    playBased,
    manualOnly,
}

internal enum class MachineImportSource {
    pinside,
}

internal enum class MachineImportMatchConfidence {
    high,
    medium,
    low,
    manual,
}
